Update Protocolo de Comunicacion.ino

Ignagx opened the pull request.

Ignagx committed:
Update Lora_RX.ino

martinbedouret commented on src/Lora_RX/Lora_RX.ino:
```diff
@@ -27,133 +27,237 @@ SFE_BMP180 pressure;
 #define DATA_TIME 500
 
 double altitud;
+bool responder = 0, haRespondido = 0, noPudoProcesar = 0, escribirDatos = 0, ejecutarAccion = 0, inicializacion = 0, error = 0, cansatReset = 0;
```
no deberías usar `true` o `false` para las variables bool?

Ignagx replied:
Uso 1s y 0s nada mas por costumbre del mplab, donde no existe el `true` y el `false`. Deberia ponerlo con eso? o lo dejo asi?

martinbedouret replied:
en C deberías usar los valores true y false. Mejor cambialo.

martinbedouret commented on src/Lora_RX/Lora_RX.ino:
```diff
@@ -27,133 +27,237 @@ SFE_BMP180 pressure;
 
-        String tiempo = LoRaData.substring(indicador10 + 1, indicador11);
+            case 3:
+                resetFunc(); // El mensaje recibido es para resetear el sistema
```
Esta función no está definida, debería ser `reset()`?

Ignagx replied:
Esta definida arriba del setup. Nose como referenciar a partes del codigo para mostrarte

martinbedouret replied:
arduino no me compila por este error 
![image](https://user-images.githubusercontent.com/8397930/186015465-dd78c505-d547-4f27-9cfc-8cebd79545e5.png)

Ignagx replied:
Si, enrealidad esta al fondo del codigo pero esta mal, deberia estar arriba del setup para que la tome. Igualmente nose si hay una funcion nativa de arduino para resetear, esa forma la saque de internet, fue lo primero que me aparecio.

martinbedouret commented on src/Lora_RX/Lora_RX.ino:
```diff
@@ -27,133 +27,237 @@ SFE_BMP180 pressure;
+        }
 
-        if (sAltitud.toDouble() >= 10)
+        if (escribirDatos)
```
En el protocolo, escribir datos es el primer byte, pero para definir completamente la acción necesitas primer y segundo byte. En este caso debemos agregar al protocolo un valor para el envío de toda la telemetria del satélite. Por ejemplo: 
```
|-----------------------|
| request/response code |
|-----------------------|
|      1    |     4     |
|-----------------------|
```
Código de solicitud/respuesta (2 bytes)
--
El primer byte indica la operación a ejecutar
0: leer datos y solicitud de información
1: escribir datos
2: ejecutar acción
3:borrar, reset
 
El segundo byte indica el elemento o acción específica
0:  se inicializan las variables de comienzo y da por empezada la misión.
1: se aborta la misión de flyback, aterrizar donde se encuentre el cansat.
4: datos: telemetria completa

Ignagx replied:
perfecto

Ignagx committed:
Update Lora_RX.ino

Ya le cambie los valores a las variables booleanas, y le agrego el sensor de calidad de aire

fpp3 committed:
Update Lora_TX.ino

Reducido un poco el tamaño del programa (95% a 86%)
Repetitividad reducida con el uso de funciones
Codigo acomodado
Prototipos agregados
Ahora se usa un puerto SPI fisico para la MicroSD (antes era por software)
Ahora funciona la telemetria local en MicroSD
Cambios en el Protocolo de Transmision

Hay fluctuaciones de latencia de transmision (+/-20ms)
De alguna forma dejo de andar el sensor de temperatura pero no el de presion
Los periodos de escritura de la MicroSD no concuerdan con lo establecido (23ms vs 60/70ms)
Hay que buscar una manera de no tener que cerrar el archivo cada vez que escribamos una linea en la MicroSD

fpp3 committed:
Add files via upload

Librerias usadas en esta rama con sus respectivas modificaciones

fpp3 committed:
Delete arduino-LoRa-STM32-master.zip

Libreria en desuso reemplazada por la estandard de arduino

fpp3 committed:
Update Lora_TX.ino

Optimizado aun mas el uso de la memoria estatica (86 a 84%)
Ahora no abrimos y cerramos el mismo archivo por cada linea que escribimos. Una vez abierto, usamos File.flush() para pasar todo a la MicroSD y liberar memoria dinamica
Los tiempos de escritura tienen una falla periodica que duplica su tiempo de escritura
Cambiamos la libreria SDFat por la libreria estandard de arduino SD (mas liviana, menos compleja)

fpp3 committed:
Add files via upload

Ignagx committed:
Update Lora_TX.ino

Ignagx committed:
Merge pull request #5 from its-villada/ffpp2003-patch-2_Bluepill_TX_gVIE-Protocol

Update Lora_TX.ino

Ignagx committed:
Update Lora_RX.ino

Ignagx committed:
Update Lora_RX.ino

Ignagx committed:
Update Lora_TX.ino

Ignagx committed:
Update Lora_RX.ino

Ignagx committed:
Update Lora_TX.ino

Ignagx committed:
Update Lora_TX.ino

Ignagx committed:
Update Lora_RX.ino

Ignagx commented:
Martin

Lo de las variables bool uso 1s y 0s solo por costumbre del mplab donde no
existe true y false. Si queres lo puedo cambiar

La funcion reset esta definida arriba del setup, si la ponia abajo no me la
tomaba. A esa funcion la saque de un foro en internet

Me parece bien lo del protocolo. En el caso desde el cansat, lo unico que
enviaba era el 1 de enviar datos y al segundo byte lo enviaba como x, osea
indefinido

On Mon, 22 Aug 2022 at 5:14 PM martin bedouret ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
> ------------------------------
>
> In src/Lora_RX/Lora_RX.ino
> <https://github.com/its-villada/mCALCAN/pull/3#discussion_r951810465>:
>
> > @@ -27,133 +27,237 @@ SFE_BMP180 pressure;
>
>  #define DATA_TIME 500
>
>
>
>  double altitud;
>
> +bool responder = 0, haRespondido = 0, noPudoProcesar = 0, escribirDatos = 0, ejecutarAccion = 0, inicializacion = 0, error = 0, cansatReset = 0;
>
>
> no deberías usar true o false para las variables bool?
> ------------------------------
>
> In src/Lora_RX/Lora_RX.ino
> <https://github.com/its-villada/mCALCAN/pull/3#discussion_r951823090>:
>
> >
>
> -        String tiempo = LoRaData.substring(indicador10 + 1, indicador11);
>
> +            case 3:
>
> +                resetFunc(); // El mensaje recibido es para resetear el sistema
>
>
> Esta función no está definida, debería ser reset()?
> ------------------------------
>
> In src/Lora_RX/Lora_RX.ino
> <https://github.com/its-villada/mCALCAN/pull/3#discussion_r951870151>:
>
> >
>
> -        if (sAltitud.toDouble() >= 10)
>
> +        if (escribirDatos)
>
>
> En el protocolo, escribir datos es el primer byte, pero para definir
> completamente la acción necesitas primer y segundo byte. En este caso
> debemos agregar al protocolo un valor para el envío de toda la telemetria
> del satélite. Por ejemplo:
>
> |-----------------------|
>
> | request/response code |
>
> |-----------------------|
>
> |      1    |     4     |
>
> |-----------------------|
>
>
> Código de solicitud/respuesta (2 bytes)
>
> El primer byte indica la operación a ejecutar
> 0: leer datos y solicitud de información
> 1: escribir datos
> 2: ejecutar acción
> 3:borrar, reset
>
> El segundo byte indica el elemento o acción específica
> 0: se inicializan las variables de comienzo y da por empezada la misión.
> 1: se aborta la misión de flyback, aterrizar donde se encuentre el cansat.
> 4: datos: telemetria completa
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/its-villada/mCALCAN/pull/3#pullrequestreview-1081011149>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A2LUDKCX7OWV5MZ4GFXKJODV2PNTJANCNFSM57FRPMTQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
